tests/TestCase.php

```diff
@@ -24,7 +23,25 @@
  */
 abstract class TestCase extends BaseTestCase
 {
-    use ExpectDeprecationTrait;
+    /** @var null|callable */
+    private $previouslyDefinedErrorHandler;
+
+    /** @var array<string> */
+    private array $expectedDeprecations = [];
+
+    /** @var array<string> */
+    private array $actualDeprecations = [];
+
+    protected function tearDown(): void
+    {
+        if (null !== $this->previouslyDefinedErrorHandler) {
```

Comment:
is anything around https://github.com/sebastianbergmann/phpunit/issues/5532 helping us to avoid local implementation?

Reply:
I wasn't able to figure anything out that would still work with PHP 7.4 and PHPUnit 9.